lte/gateway/c/core/oai/tasks/amf/Registration.cpp

```diff
@@ -220,16 +220,35 @@ int amf_proc_registration_request(amf_ue_ngap_id_t ue_id,
     }
   }
 
+  /* Implicit deregistartion of existing context should be triggered if
+   * same TMSI used and trigger fresh registration request*/
+  if (ies->guti) {
+    amf_app_desc_t* amf_app_desc_p = get_amf_nas_state(false);
+    ue_m5gmm_context_s* guti_ue_mm_ctx = NULL;
+
+    guti_ue_mm_ctx =
+        amf_ue_context_exists_guti(&amf_app_desc_p->amf_ue_contexts, ies->guti);
+    if (guti_ue_mm_ctx) {
+      old_ue_id = guti_ue_mm_ctx->amf_ue_ngap_id;
+      if ((guti_ue_mm_ctx->mm_state == REGISTERED_CONNECTED) ||
+          (guti_ue_mm_ctx->mm_state == REGISTERED_IDLE)) {
+        amf_nas_proc_implicit_deregister_ue_ind(old_ue_id);
+      }
+    }
+  }
   if (!(is_nas_specific_procedure_registration_running(
           &ue_m5gmm_context->amf_context))) {
     amf_proc_create_procedure_registration_request(ue_m5gmm_context, ies);
   } else {
     /* Update the GUTI */
-    if (ies->guti) {
-      nas_amf_registration_proc_t* registration_proc =
-          get_nas_specific_procedure_registration(
-              &(ue_m5gmm_context->amf_context));
-
+    nas_amf_registration_proc_t* registration_proc =
+        get_nas_specific_procedure_registration(
+            &(ue_m5gmm_context->amf_context));
+    if (registration_proc == NULL) {
```

Comment:
If registration_proc is NULL shouldn`t this function return with ERROR? And checks in line 247 and 248 can be combined

Reply:
Handled the changes.

Reply:
done